fastlane_core/lib/fastlane_core/configuration/config_item.rb

```diff
@@ -102,14 +102,12 @@ def initialize(key: nil, env_name: nil, description: nil, short_option: nil, def
       @allow_shell_conversion = (type == :shell_string)
     end
 
-    # This will raise an exception if the value is not valid
     def verify!(value)
-      UI.user_error!("Invalid value '#{value}' for option '#{self}'") unless valid?(value)
-      true
+      true if valid?(value)
```

Comment:
Yes, technically the correct thing to do would be to raise the exception here (method with the `!`), and only return booleans in `valid?`. However this would also mean that we can't show specific error messages that are helpful for the user to understand. Maybe we should just have one method that raises the exception?